compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -2223,6 +2223,24 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                                         Err(LitToConstError::TypeError) => todo!(),
                                     }
                                 }
+
+                                hir::ExprKind::Path(hir::QPath::Resolved(
+                                    _,
+                                    &hir::Path {
+                                        res: Res::Def(DefKind::ConstParam, def_id), ..
+                                    },
+                                )) => {
```

Comment:
Note that we're missing a call to `prohibit_generic_args` here. Right now, we permit:

```rs
pattern_type!(u32 is START::<(), i32, 2>..=END::<_>)
```

While we could add this check, this just reinforces oli's point: https://github.com/rust-lang/rust/pull/123689#discussion_r1558148361, [Zulip discussion](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/pattern.20type.20HIR.20nodes/near/432410768).

Reply:
Just noticed this while reviewing an unrelated PR.